plugins/modules/aws_eks_fargate_profile.py

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/python
+# Copyright (c) 2022 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: aws_eks_fargate_profile
+version_added: 1.0.0
+short_description: Manage EKS Fargate Profile
+description:
+    - Manage EKS Fargate Profile
+author: Tiago Jarra (@tjarra)
+options:
+  name:
+    description: Name of EKS Fargate Profile
+    required: True
+    type: str
+  cluster_name:
+    description: Name of EKS Cluster
+    required: True
+    type: str
+  role_arn:
+    description: ARN of IAM role used by the EKS cluster
+    type: str
+  subnets:
+    description: list of subnet IDs for the Kubernetes cluster
+    type: list
+    elements: str
+  selectors:
+    description: A list of selectors to use in fargate profile
+    type: list
+    elements: dict
+    suboptions:
+      namespace:
+        description: A namespace used in fargate profile
+        type: str
+      labels:
+        description: A dictionary of labels used in fargate profile
+        type: dict
+        elements: str
+  state:
+    description: Create or delete the Fargate Profile
+    choices:
+      - absent
+      - present
+    default: present
+    type: str
+  tags:
+    description: A dictionary of resource tags
+    type: dict
+  purge_tags:
+    description:
+      - Purge existing tags that are not found in the cluster
+    type: bool
+    default: true
+  wait:
+    description: >-
+      Specifies whether the module waits until the profile is created or deleted before moving on.
+    type: bool
+    default: false
+  wait_timeout:
+    description: >-
+      The duration in seconds to wait for the cluster to become active. Defaults
+      to 1200 seconds (20 minutes).
+    default: 1200
+    type: int
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+
+'''
+
+EXAMPLES = r'''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Create an EKS Fargate Profile
+  community.aws.aws_eks_fargate_profile:
+    name: test_fargate
+    cluster_name: test_cluster
+    role_arn: my_eks_role
+    subnets:
+      - subnet-aaaa1111
+    selectors:
+      - namespace: nm-test
+        labels:
+          - label1: test
+    state: present
+    wait: yes
+
+- name: Remove an EKS cluster
+  community.aws.aws_eks_cluster:
+    name: test_fargate
+    clusterName: test_cluster
+    wait: yes
+    state: absent
```

Comment:
I'm not sure if this is necessary for the example section ...

Reply:
I think it's quite nice sometimes to see the barebones removal example, so that it's clear at a glance

Reply:
I kept the example and corrected the name which was wrong